src/sql/workbench/contrib/notebook/browser/notebookExplorer/notebookSearchView.ts

```diff
@@ -0,0 +1,1332 @@
+
+	private onDidChangeWorkbenchState(): void {
+		if (this.contextService.getWorkbenchState() !== WorkbenchState.EMPTY && this.searchWithoutFolderMessageElement) {
+			dom.hide(this.searchWithoutFolderMessageElement);
+		}
+	}
+
+	public onDidNotebooksOpenState(): void {
+		if (this.contextKeyService.getContextKeyValue('bookOpened') && this.searchWithoutFolderMessageElement) {
+			dom.hide(this.searchWithoutFolderMessageElement);
+		}
+	}
+
+	renderBody(parent: HTMLElement): void {
+		super.renderBody(parent);
+		this.container = dom.append(parent, dom.$('.search-view'));
+
+		this.messagesElement = dom.append(this.container, $('.messages'));
+		if (this.contextService.getWorkbenchState() === WorkbenchState.EMPTY) {
+			this.showSearchWithoutFolderMessage();
+		}
+
+		this.createSearchResultsView(this.container);
+
+		this._register(this.viewModel.searchResult.onChange((event) => this.onSearchResultsChanged(event)));
+		this._register(this.onDidChangeBodyVisibility(visible => this.onVisibilityChanged(visible)));
+
+		// initialize as collapsed viewpane
+		this.setExpanded(false);
+	}
+
+	private onVisibilityChanged(visible: boolean): void {
+		this.viewletVisible.set(visible);
+		if (visible) {
+			if (this.changedWhileHidden) {
+				// Render if results changed while viewlet was hidden - #37818
+				this.refreshAndUpdateCount().catch(errors.onUnexpectedError);
+				this.changedWhileHidden = false;
+			}
+
+			if (this.updatedActionsWhileHidden) {
+				// The actions can only run or update their enablement when the view is visible,
+				// because they can only access the view when it's visible
+				this.updateActions();
+				this.updatedActionsWhileHidden = false;
+			}
+		}
+
+		// Enable highlights if there are searchresults
+		if (this.viewModel) {
+			this.viewModel.searchResult.toggleHighlights(visible);
+		}
+	}
+
+	/**
+	 * Warning: a bit expensive due to updating the view title
+	 */
+	public updateActions(): void {
+		if (!this.isVisible()) {
+			this.updatedActionsWhileHidden = true;
+		}
+
+		for (const action of this.actions) {
+			action.update();
+		}
+
+		this.cancelAction.update();
+		this.toggleCollapseAction.update();
+
+		super.updateActions();
+	}
+
+	private async onSearchResultsChanged(event?: IChangeEvent): Promise<void> {
+		if (this.isVisible()) {
+			return this.refreshAndUpdateCount(event);
+		} else {
+			this.changedWhileHidden = true;
+		}
+	}
+
+	private async refreshAndUpdateCount(event?: IChangeEvent): Promise<void> {
+		this.updateSearchResultCount(this.viewModel.searchResult.query!.userDisabledExcludesAndIgnoreFiles);
+		return this.refreshTree(event);
+	}
+
+	async refreshTree(event?: IChangeEvent): Promise<void> {
+		const collapseResults = this.searchConfig.collapseResults;
+		if (!event || event.added || event.removed) {
+			// Refresh whole tree
+			if (this.searchConfig.sortOrder === SearchSortOrder.Modified) {
+				// Ensure all matches have retrieved their file stat
+				await this.retrieveFileStats()
```

Comment:
if you're already awaiting it, then just continue this action on the bottom, not inside a .then()

Reply:
that's something I had to add since PR Validation build kept failing on these lines with floating promises hygiene errors.